### PR TITLE
Replace and unify unreliable Article page AV Player tests

### LINKED ISSUE #6121: Intermittent failures in articles AV playback e2e tests
**Describe the bug**
We see intermittent failures of the articles media playback tests in our e2es such as:
```
articles - /news/articles/cj7xrxz0e8zo - Canonical.Canonical Tests for news articles.Media Player: Canonical

CypressError: Timed out retrying: cy.its() errored because the property: 'embeddedMedia' does not exist on your subject.
```
Playback tests for media asset and live radio pages were removed in https://github.com/bbc/simorgh/pull/6001 which means that `articles` is the only page type that still runs these tests.

**To Reproduce**
Observe recent test failures on CodeBuild (e.g. `7b6322fc-7de1-4570-aa5d-aee5e22cffd0`)

**Expected behaviour**
There should be no intermittently failing tests.

**Screenshots**
Screenshot 2020-04-07 at 14 31 30

**Testing notes**

- [x] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by simonsinclair**
Resolves https://github.com/bbc/simorgh/issues/6121.

### Overall change:
Improve the reliability of the Article page integration test suite by replacing and unifying unreliable AV Player tests.

#### Unreliable tests:
- Canonical `plays media when a user clicks play`
- AMP `should autoplay`
- Canonical `should have subtitles set to enabled by default`

These tests all attempt to read the global `window` object of a cross-origin iframe to determine the state of the video player. This has proven unreliable and probably ventures into testing the SMP player, rather than Simorgh. The Media team have long [replaced](https://github.com/bbc/simorgh/pull/6001) their equivalent tests.

### Code changes:
- Replace and unify:
  - AMP `should autoplay`
  - Canonical `plays media when a user clicks play`
    - With `should render an iframe with a valid URL | when a user clicks play`, which checks an iframe is visible and its URL is an HTML page that responds with a 200 status code.
- Remove `should have subtitles set to enabled by default`
  - This test not only relies on reading the global `window` object of a cross-origin iframe, but also reads a non-public SMP property to make its assertion.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

## Testing:

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- ~~[ ] This PR requires manual testing~~

### CODE REVIEW

File: cypress/integration/pages/articles/testsForAMPOnly.js [hunk 1]
```diff
@@ -55,33 +57,18 @@ export const testsThatFollowSmokeTestConfigForAMPOnly = ({
...
-            );
+        it('should render an iframe with a valid URL when a user clicks play', () => {
+          cy.request(`${Cypress.env('currentPath')}.json`).then(({ body }) => {
+            const media = getBlockData('video', body);
+
+            if (media && media.type === 'video') {
+              const { lang } = appConfig[service][variant];
+              const embedUrl = `${getVideoEmbedUrl(body, lang)}/amp`;
+              cy.testResponseCodeAndType(embedUrl, 200, 'text/html');
+              cy.get(`amp-iframe[src="${embedUrl}"]`).should('be.visible');
```
Comment (karinathomasbbc) [inline cypress/integration/pages/articles/testsForAMPOnly.js]: Would it make sense to swap these lines - check if the iframe is visible first, before checking the endpoint?

File: cypress/integration/pages/articles/testsForCanonicalOnly.js [hunk 2]
```diff
@@ -160,57 +162,25 @@ export const testsThatFollowSmokeTestConfigForCanonicalOnly = ({
...
+            if (media && media.type === 'video') {
+              const { lang } = appConfig[service][variant];
+              const embedUrl = getVideoEmbedUrl(body, lang);
+              cy.testResponseCodeAndType(embedUrl, 200, 'text/html');
+              cy.get(
+                'div[class*="StyledVideoContainer"] button[class*="StyledPlayButton"]',
+              )
+                .click()
+                .then(() => {
+                  cy.get(`iframe[src="${embedUrl}"]`).should('be.visible');
```
Comment (karinathomasbbc) [inline cypress/integration/pages/articles/testsForCanonicalOnly.js:178]: Is it also worth doing the embed URL check here too?

`cy.testResponseCodeAndType(embedUrl, 200, 'text/html');`

Comment (karinathomasbbc) [reply cypress/integration/pages/articles/testsForCanonicalOnly.js:178]: Oops, you already do!

**Review by karinathomasbbc — APPROVED** [review]

File: cypress/integration/pages/articles/testsForAMPOnly.js (hunk 1 above)
Comment (simonsinclair) [reply cypress/integration/pages/articles/testsForAMPOnly.js]: Yes, I think it probably would - I'll swap them!

Comment (karinathomasbbc) [reply cypress/integration/pages/articles/testsForAMPOnly.js]: Thanks :)

**Review by FK78 — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
Had a chat with Simon regarding this PR, simon has run the non-smoke test suite and everything’s passed, I am happy for this ro be merged, doesn't require any manual testing..

**Pull request merged**